### PR TITLE
Add view background color matcher

### PR DESCRIPTION
**Pull request by anton46**
Hi Guys, I've added a view background color matcher for check the android view background based on color resource or color code. If there is something wrong just let me know and I am happy to discuss.

### CODE REVIEW

File: kakao/src/main/kotlin/com/agoda/kakao/Matchers.kt [hunk 1]
```diff
@@ -213,4 +220,37 @@ class DrawableMatcher(@DrawableRes private val resId: Int = -1, private val draw
...
+            }
+
+            it.background != null &&
+            it.background.current is ColorDrawable &&
+            (it.background.current as ColorDrawable).color == expectedColor
+        }?: false
+    }
+
+    override fun describeTo(description: Description) {
+        description.appendText("with background color resource or provided color code")
```
Comment (Alviere) [inline kakao/src/main/kotlin/com/agoda/kakao/Matchers.kt]: You should append here actual values that were passed to the matcher. It makes debugging much easier.

**Comment by cdsap** [comment]
@anton46 thanks for the contribution :) 
Two things

- There is one of your users that needs to sign the CLA
- Can you squash the commits ?

**Comment by anton46** [comment]
@Alviere Thank you for the prompt answer. The`hasBackgroundColor()` should be pass a function or the color information(res/string)?

**Comment by anton46** [comment]
@cdsap Yep, I will update the commit.

**Comment by Alviere** [comment]
@anton46 it should be overloaded function. `hasBackgroundColor(@ColorRes resId: Int)` and `hasBackgroundColor(color: String)`

**Pull request closed without merging**